### ISSUE TITLE
Jsnapy not using kwargs in snapshot file name when using RPC

### ISSUE THREAD

**Issue by alejandrosalinas** [Juniper/jsnapy#230]
When using RPC JSnapy creates a snapshot file using a slug of the RPC call and not considering the arguments. This makes Jsnapy overwrite one result with the other. For example

```
test_rpc_bgp_prefix1_route:
  - rpc: get-route-information
  - kwargs:
      destination: 10.1.2.0/24
      exact: True
      active-path: True
  - iterate:
      xpath: '//route-table/rt/rt-entry/nh/selected-next-hop'
      tests:
        - is-equal: ../to, 10.1.1.1
          err: "Test Failed!! preferred next hop is <{{post['../to']}}>"
          info: "Test succeeded!! preferred next hop is <{{post['../to']}}>"

test_rpc_bgp_prefix2_route:
  - rpc: get-route-information
  - kwargs:
      destination: 0.0.0.0/0
      exact: True
      active-path: True
  - iterate:
      xpath: '//route-table/rt/rt-entry/nh/selected-next-hop'
      tests:
        - is-equal: ../to, 10.1.2.1
          err: "Test Failed!! preferred next hop is <{{post['../to']}}>"
          info: "Test succeeded!! preferred next hop is <{{post['../to']}}>"
```
Would create two snapshots with the same name, overwriting one and yielding incorrect test results for one of them.

**Comment by eeishaan** [comment]
@alejandrosalinas Thank you for bringing this to our notice. This is a limitation on our part.
I understand that using kwargs returns a concise snapshot. I would suggest you to write your test in the following manner:

```yaml
test_rpc_bgp_prefix_route:
  - rpc: get-route-information
  - iterate:
      xpath: //route-table/rt[rt-destination='10.0.0.0/10']/rt-entry/nh/selected-next-hop
      tests:
        - is-equal: ../to, 10.1.1.1
          err: "Test Failed!! preferred next hop is <{{post['../to']}}>"
          info: "Test succeeded!! preferred next hop is <{{post['../to']}}>"          
  - iterate:
      xpath: //route-table/rt[rt-destination='0.0.0.0/0']/rt-entry/nh/selected-next-hop
      tests:
        - is-equal: ../to, 10.1.2.1
          err: "Test Failed!! preferred next hop is <{{post['../to']}}>"
          info: "Test succeeded!! preferred next hop is <{{post['../to']}}>"
``` 

I am no expert of `get-route-information` rpc. But this is logically equivalent to your test cases. You can always split these up into two test cases. 
There is an existing work item #175 where we take care of duplicate rpc/command so that they are executed only once. Looks like we may have to redefine "duplicate".
